easyfinance.client/src/app/features/user/detail-user/detail-user.component.ts

```diff
@@ -103,6 +117,67 @@ export class DetailUserComponent implements OnInit {
     });
   }
 
+  sanitizeMessage(message: string): SafeHtml {
+    return this.sanitizer.bypassSecurityTrustHtml(message);
+  }
+  
+  openDeleteDialog(): void {
+    this.deleteError = null;
+    this.deleteModalError = null;
+
+    const dialogRef: MatDialogRef<any> = this.dialog.open(this.deleteDialog, {
+      width: '400px',
+    });
+    const token = this.tokenService.getToken();
+    const confirmationMessage = this.tokenService.getConfirmationMessage();
+    if(confirmationMessage) this.confirmationMessage = confirmationMessage;
+    if(!token){
+    this.userService.deleteUser().subscribe({
+      next: (response: any) => {
+        if (response?.confirmationToken) {
+          this.confirmationMessage = response.confirmationMessage; 
+          this.tokenService.setToken(response.confirmationToken, response.confirmationMessage); 
+        }
+      },
+      error: (err) => {
+        console.error('Error during first deletion attempt:', err);
+        this.deleteError = 'Failed to delete account. Please try again later';      
+      },
+    });
+    }
+    
+    dialogRef.afterClosed().subscribe(result => {
+      if (result === true) {
+        this.confirmDeletion();
+      }
+    });
+  }
+
+  closeDialog(): void {
+    this.deleteError = null;
+    this.deleteModalError = null;
+    this.dialog.closeAll();
+  }
+
+  confirmDeletion(): void {
+    this.deleteModalError = null;
+    const token = this.tokenService.getToken();
+    if (token) {
+      this.userService.deleteUser(token).subscribe({
```

Comment:
```suggestion
      this.userService.deleteUser().subscribe({
```

If you accepted the suggestion to use the tokenService in the userService, we need to remove the token here.